drivers/flash/Kconfig.nrf

```diff
@@ -27,6 +27,33 @@ config SOC_FLASH_NRF_RADIO_SYNC
 	help
 	  Enable synchronization between flash memory driver and radio.
 
+config SOC_FLASH_NRF_PARTIAL_ERASE
+	bool "Nordic nRFx flash driver partial erase"
+	default n
```

Comment:
need to mask it depends on SoC's that have this feature - for these devices `NVMC_ERASEPAGEPARTIALCFG_DURATION_Msk` is defined in MDK.
Make this `depends on` these SoC directly, or
introduce another `HAS_HW_NRF_something`  Kconfig property in https://github.com/zephyrproject-rtos/zephyr/blob/master/soc/arm/nordic_nrf/Kconfig.peripherals and select it for proper SoC's (e.g. her https://github.com/zephyrproject-rtos/zephyr/blob/master/soc/arm/nordic_nrf/nrf52/Kconfig.soc). make this config dependes on this new kconfig keyword.

Reply:
@anangl ^^ this is issue I mentioned today on the daily meeting.